analyzer/codechecker_analyzer/analyzer_context.py

```diff
@@ -71,13 +71,20 @@ def __init__(self):
         # Original caller environment of CodeChecker for external binaries
         self.__original_env = None
 
-        self.logger_lib_dir_path = os.path.join(
-            self._data_files_dir_path, 'ld_logger', 'lib')
-
-        if not os.path.exists(self.logger_lib_dir_path):
-            self.logger_lib_dir_path = os.path.join(
+        # Find the path which has the architectures for the built ld_logger
+        # shared objects.
+        ld_logger_path = Path(self._data_files_dir_path, 'ld_logger', 'lib')
+        if not ld_logger_path.is_dir():
+            ld_logger_path = Path(
                 self._lib_dir_path, 'codechecker_analyzer', 'ld_logger', 'lib')
 
+        # Add the ld_logger_path and all children (architecture) paths to be
+        # later used in the LD_LIBRARY_PATH environment variable during
+        # logging of compiler invocations.
+        self.logger_lib_dir_path = ':'.join(
+            [str(ld_logger_path)] +
```

Comment:
I think we dont need to add CodeChecker/ld_logger/lib/
and then we can also remove the  'i486' 'i586' 'i686' subdirectories from analyzer/tools/build-logger/Makefile line 79
`  for x86dir in 'i386' 'i486' 'i586' 'i686'; do \`